Assets/Tests/PlayModeTests/Scenarios/UP_UserProfile.cs

```diff
@@ -0,0 +1,410 @@
+            };
+
+            return expectedUserDetails;
+        }
+
+        private void SendSameData()
+        {
+            Countly.Instance.UserProfile.SetProperty("a12345", "1");
+            Countly.Instance.UserProfile.SetProperty("a12345", "2");
+            Countly.Instance.UserProfile.SetProperty("a12345", "3");
+            Countly.Instance.UserProfile.SetProperty("a12345", "4");
+        }
+
+        private Dictionary<string, object> ExpectedSameData()
+        {
+            // Example expected user details for validation
+            var expectedUserDetails = new Dictionary<string, object>
+            {
+                { "a12345", "4" }
+            };
+
+            return expectedUserDetails;
+        }
+
+        // UserProfile calls in Countly.Instance.UserProfile
+        // We initialize SDK without consent requirement, check the RQ and EQ sizes, then call public methods
+        // Methods we call should record nothing in the RQ and EQ
+        [Test]
+        public void UP_200_CNR_A()
+        {
+            CountlyConfiguration config = TestUtility.CreateBaseConfig()
+                .SetRequiresConsent(false);
+            Countly.Instance.Init(config);
+            TestUtility.ValidateRQEQSize(Countly.Instance, 1, 0);
+
+            SendUserData();
+            SendUserProperty();
+            TestUtility.ValidateRQEQSize(Countly.Instance, 1, 0);
+        }
+
+        // UserProfile calls in Countly.Instance.UserProfile
+        // We initialize SDK with consent requirement and give consents, check the RQ and EQ sizes, then call public methods
+        // Methods we call should record nothing in the RQ and EQ
+        [Test]
+        public void UP_201_CR_CG_A()
+        {
+            Consents[] consent = new Consents[] { Consents.Crashes, Consents.Events, Consents.Clicks, Consents.StarRating, Consents.Views, Consents.Users, Consents.Push, Consents.RemoteConfig, Consents.Location, Consents.Feedback, Consents.Sessions };
+            CountlyConfiguration config = TestUtility.CreateBaseConfigConsent(consent);
+            Countly.Instance.Init(config);
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);
+
+            SendUserData();
+            SendUserProperty();
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);
+        }
+
+        // UserProfile calls in Countly.Instance.UserProfile
+        // We initialize SDK with consent requirement and give no consent, check the RQ and EQ sizes, then call public methods
+        // Methods we call should record nothing in the RQ and EQ
+        [Test]
+        public void UP_202_CR_CNG_A()
+        {
+            Consents[] consent = new Consents[] { };
+            CountlyConfiguration config = TestUtility.CreateBaseConfigConsent(consent);
+            Countly.Instance.Init(config);
+            TestUtility.ValidateRQEQSize(Countly.Instance, 2, 0);

```

Comment:
if there is no consent what are those two?
location and consent right?

Reply:
yup sir